python/ray/tests/test_basic.py

```diff
@@ -69,6 +69,35 @@ def consume(refs):
     assert len(pids) <= 2, pids
 
 
+def test_not_release_resource(shutdown_only):
+    # Test to make sure we don't release CPU
+    # resource if the object is already fetched.
+    ray.init(num_cpus=1)
+
+    @ray.remote
+    def task1():
+        return [1] * (1024 * 1024)
+
+    o1 = task1.remote()
+
+    @ray.remote
+    def task2(*args, **kwargs):
+        # ray.get here should not release
+        # CPU resource since the object is already
+        # available in args[0]
+        assert args[0] == ray.get(kwargs["o"][0])
```

Comment:
Since the object is part of `args`, `CoreWorker::GetAndPinArgsForExecutor` will be called and `OBJECT_IN_PLASMA` is supposed to be added to memory store so that later on when we call `ray.get(kwargs["o"][0])`, we will realize that the object is in plasma and won't release CPU resource.

Before this PR, `CoreWorker::GetAndPinArgsForExecutor` tries to add `OBJECT_IN_PLASMA` to memory store but it's a no-op because there is no reference in the reference table. Later on when we call `ray.get(kwargs["o"][0])`, it cannot find the object in the memory store so it will release CPU resource and asynchronously resolve the object from the owner.

The issue of releasing CPU resources unnecessarily (there is no deadlock in this case) is that we will run too many tasks and each of them takes some memory (even when they are blocked on ray.get()) and eventually causes OOM.